### PR TITLE
T004 media insertion

### PR DESCRIPTION
**Pull request by MaxDangVu136**
Issue: no video implemented. 
Solution: implemented YouTube video feature (minimal viable solution). 
Risk: may override some of Kenneth's previous implementation.

### CODE REVIEW

**Review by johnchen383 — CHANGES_REQUESTED** [review]
Please remove .vscode from being pushed to remote repo - it doesn't need to be there.

Alos, node-sass was removed as a package. This needs to be there.

Remove video-react if not using it.

Not too sure why tsconfig changed. Please advise if its needed. Else, react-jsx should be the proper field.

**Comment by johnchen383** [comment]
Also, please try to remove package-lock.json... im assuming a npm command was used at some point but we dont want duplicate files

**Comment by johnchen383** [comment]
In general, try to make your components widgets first and then we can merge to main. Good work though!

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "react-dom": "^17.0.2",
     "react-router-dom": "^6.2.1",
     "react-scripts": "5.0.0",
+    "react-video": "^1.5.3",
```
Comment (johnchen383) [inline package.json]: remove dependency

File: package.json [hunk 2]
```diff
@@ -45,5 +46,8 @@
   },
   "resolutions": {
     "mini-css-extract-plugin": "2.4.5"
+  },
```
Comment (johnchen383) [inline package.json]: remove dependency

File: src/styles/styles.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.video-responsive {
```
Comment (johnchen383) [inline src/styles/styles.scss]: try for a better name than `styles.scss`. How does this differ from youtubevideo.scss

File: src/styles/youtube_video.scss [hunk 4]
```diff
@@ -0,0 +1,5 @@
+.cust_video {
```
Comment (johnchen383) [inline src/styles/youtube_video.scss]: remove underscore for consistent naming.. we use camelcase for styles

File: src/widgets/YouTubeVideo.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+// Source code: https://dev.to/bravemaster619/simplest-way-to-embed-a-youtube-video-in-your-react-app-3bk2
```
Comment (johnchen383) [inline src/widgets/YouTubeVideo.tsx]: If poss, would remove capitalisation of T

**Review by johnchen383 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (johnchen383) [reply package.json]: dev sass dependecy to clarify

File: package.json (hunk 1 above)
Comment (johnchen383) [reply package.json]: react-video

**Review by johnchen383 — CHANGES_REQUESTED** [review]
As commented

File: src/styles/youtube-video.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
+.video-responsive {
```
Comment (johnchen383) [inline src/styles/youtube-video.scss]: please use camelCase for .scss files

File: src/widgets/Youtube-Video.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import "../styles/youtube-video.scss";
```
Comment (johnchen383) [inline src/widgets/Youtube-Video.tsx]: please use PascalCase for .tsx files

**Review by johnchen383 — CHANGES_REQUESTED** [review]
Just change the naming conventions and then I'll run and give it a go. But looking good!

File: public/index.html [hunk 8]
```diff
@@ -12,6 +12,7 @@
   <!-- Add fonts here from Google Fonts -->
   <link rel="preconnect" href="https://fonts.googleapis.com">
   <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link rel="stylesheet" href="https://video-react.github.io/assets/video-react.css"/>
```
Comment (johnchen383) [inline public/index.html]: Remove import plz

**Review by johnchen383 — CHANGES_REQUESTED** [review]
As commented

**Review by KennethZhu904 — APPROVED** [review]
LGTM after changes requested by John! :)

**Review by johnchen383 — APPROVED** [review]
LGTM. Thanks for all the hard work!

**Pull request merged**